frontend/src/scenes/apps/ErrorDetailsModal.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from 'react'
+import { useActions, useValues } from 'kea'
+import { AppMetricErrorDetail, appMetricsSceneLogic } from './appMetricsSceneLogic'
+import { LemonSkeleton } from 'lib/components/LemonSkeleton'
+import { TZLabel } from 'lib/components/TimezoneAware'
+import { LemonLabel } from 'lib/components/LemonLabel/LemonLabel'
+import { LemonButton } from 'lib/components/LemonButton'
+import { IconChevronLeft, IconChevronRight, IconUnfoldLess, IconUnfoldMore } from 'lib/components/icons'
+import { LemonModal } from 'lib/components/LemonModal'
+import { CodeSnippet, Language } from '../ingestion/frameworks/CodeSnippet'
+
+export function ErrorDetailsModal(): JSX.Element {
+    const { errorDetails, errorDetailsModalError, errorDetailsLoading } = useValues(appMetricsSceneLogic)
+    const { closeErrorDetailsModal } = useActions(appMetricsSceneLogic)
+    const [page, setPage] = useState(0)
+
+    const activeErrorDetails: AppMetricErrorDetail = errorDetails[page]
+
+    return (
+        <LemonModal
+            isOpen={!!errorDetailsModalError}
+            onClose={closeErrorDetailsModal}
+            title={errorDetailsModalError}
+            width={'min(50vw, 80rem)'}
+            description={<span>{activeErrorDetails?.error_details?.error.message?.substring(0, 200)}</span>}
+            footer={
+                <div className="flex items-center justify-end gap-1 h-">
+                    {errorDetailsLoading ? (
+                        <LemonSkeleton className="1-10" />
+                    ) : (
+                        <>
+                            <span>
+                                {page + 1} of {errorDetails.length} sample{errorDetails.length > 1 ? 's' : ''}
+                            </span>
+                            <LemonButton
```

Comment:
I'm surprised we have to define these manually here

Reply:
You mean pagination? There wasn't a case with pagination-in-modal before and the existing components would not look nice. This is relatively small amount of code to achieve the same since we're fetching everything in one go.